### PR TITLE
Update build.yml

### PR DESCRIPTION
**Pull request by castrojo**
Add cosign to sign our artifacts

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -90,7 +94,21 @@ jobs:
           password: ${{ env.REGISTRY_PASSWORD }}
           extra-args: |
             --disable-content-trust
-
+      # Sign container
+      - uses: sigstore/cosign-installer@main
```
Comment (p5) [inline .github/workflows/build.yml:98]: How does this work?  

I would have expected this to edit the image in some way, so thought signing would happen before pushing it to GHCR.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -90,7 +94,21 @@ jobs:
...
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Sign container image
+        run: |
+          cosign sign ${{ steps.registry_case.outputs.lowercase }}/${{ env.IMAGE_NAME }}@${TAGS}
```
Comment (p5) [inline .github/workflows/build.yml:107]: Is this able to iterate over all tags?  Or do we need to repeat this command for every tag we're pushing?

**Review by p5 — COMMENTED** [review]
Couple questions, but looks like a great addition.

Also, a README update explaining how to validate the container signing might be useful  :smile:

File: .github/workflows/build.yml (hunk 2 above)
Comment (p5) [inline .github/workflows/build.yml:107]: This pushes a completely new image with the signed artefact, not updates existing ones.
I think you need to call this multiple times for each tag that should be signed.

image

**Review by castrojo — COMMENTED** [review]
Add a readme snippet

**Comment by castrojo** [comment]
Ok I can't add the output of the result to the readme until this lands so that'll go in a followup PR and also I'll have instructions on how others can set it up.

**Pull request merged**